gui/test/app/tinystan/mocking/WasmModule.ts

```diff
@@ -0,0 +1,82 @@
+import { vi } from "vitest";
+
+import type { internalTypes } from "../../../../src/app/tinystan/types";
+type ptr = internalTypes["ptr"];
+type model_ptr = internalTypes["model_ptr"];
+type cstr = internalTypes["cstr"];
+
+export type ModuleSettings = {
+  returnCode: number;
+  paramNames: string;
+  numFreeParams: number;
+  modelFails?: boolean;
+};
+
+const defaultModuleSettings: ModuleSettings = {
+  returnCode: 0,
+  paramNames: "foo,bar,baz",
+  numFreeParams: 3,
+  modelFails: false,
+};
+
+// 128MB heap, filled with increasing numbers.
+// This allows us to check where values came from
+// relative to each other
+const fakeHeap = new Float64Array(16 * 1024 * 1024);
+for (let i = 0; i < fakeHeap.length; i++) {
+  fakeHeap[i] = i;
+}
+
+export const mockModule = (p: Partial<ModuleSettings>) => {
+  const {
+    returnCode,
+    numFreeParams: numParams,
```

Comment:
Note the automatic aliasing VSCode "helpfully" did for you here (unless that was intentional)

(VSCode likes to Help(tm) when you use the F2/context-aware rename for object members that get unpacked)

Reply:
Yeah this is the kind of "deeper" refactoring I was trying to ask you if I could have vscode do...